conda_env/env.py

```diff
@@ -144,11 +145,17 @@ def from_yaml(yamlstr, **kwargs):
 
 
 def from_file(filename):
-    if not os.path.exists(filename):
+    if any(filename.startswith(prefix) for prefix in ("https://", "http://")):
+        try:
+            yamlstr = urllib.request.urlopen(filename).read()
```

Comment:
Instead of `urllib`, we should import `from conda.gateways.connection.session import CondaSession`, which is a `requests` `Session`. That hooks into our token and auth management, proxy configuration, etc. So something like

```
response = CondaSession().get(filename)
response.raise_for_status()
yamlstr = response.text
```

Reply:
Ah, that's nice.  Done.